sdk/src/Services/DynamoDBv2/Custom/DataModel/BatchGet.cs

```diff
@@ -27,318 +25,295 @@
 
         /// <summary>
         /// If set to true, a consistent read is issued. Otherwise eventually-consistent is used.
         /// </summary>
-        public bool ConsistentRead { get; set; }
-
-        #endregion
+        /// <remarks>
+        /// Refer to the <see href="https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/HowItWorks.ReadConsistency.html">
+        /// Read Consistency</see> topic in the DynamoDB Developer Guide for more information.
+        /// </remarks>
+        bool ConsistentRead { get; set; }
+    }
 
+    /// <summary>
+    /// Represents a generic interface for retrieving a batch of items
+    /// from a single DynamoDB table
+    /// </summary>
+    public interface IBatchGet<T> : IBatchGet
+    {
+        /// <summary>
+        /// List of generic results retrieved from DynamoDB.
+        /// </summary>
+        /// <remarks>
+        /// This is only populated after a call to Execute.
+        /// </remarks>
+        List<T> Results { get; }
 
-        #region Protected methods
+        /// <summary>
+        /// Add a single item to get, identified by its hash primary key.
+        /// </summary>
+        /// <param name="hashKey">Hash key of the item to get</param>
+        void AddKey(object hashKey);
 
         /// <summary>
-        /// Executes a server call to batch-get the items requested.
-        /// Populates Results with the retrieved items.
+        /// Add a single item to get, identified by its hash-and-range primary key.
         /// </summary>
-        internal protected abstract void ExecuteHelper();
+        /// <param name="hashKey">Hash key of the item to get</param>
+        /// <param name="rangeKey">Range key of the item to get</param>
+        void AddKey(object hashKey, object rangeKey);
 
-#if AWS_ASYNC_API
         /// <summary>
-        /// Executes an asynchronous server call to batch-get the items requested.
-        /// Populates Results with the retrieved items.
+        /// Add a single item to get.
         /// </summary>
-        internal protected abstract Task ExecuteHelperAsync(CancellationToken cancellationToken);
-#endif
-        #endregion
+        /// <param name="keyObject">Object key of the item to get</param>
+        void AddKey(T keyObject);
 
+        /// <summary>
+        /// Creates a MultiTableBatchGet object that is a combination
+        /// of the current BatchGet and the specified BatchGets
+        /// </summary>
+        /// <param name="otherBatches">Other BatchGet objects</param>
+        /// <returns>
+        /// An interface with the ability to perform MultiTableBatchGet operations.
+        /// </returns>
+        IMultiTableBatchGet Combine(params IBatchGet[] otherBatches);
+    }
 
-        #region Internal methods
+    /// <summary>
+    /// Represents a non-generic object for retrieving a batch of items
+    /// from a single DynamoDB table
+    /// </summary>
+    public abstract class BatchGet
```

Comment:
This should be implementing `IBatchGet`.

Reply:
Done